### PR TITLE
test: batch state coverage edges

### PR DESCRIPTION
**Pull request by danielraffel**
Batched state/parameter Codecov coverage work. This keeps the state-related tranches together in one GitHub-hosted CI run instead of splitting them across many small PRs.

Coverage/fix areas:
- StateStore deserialize clamp, bad magic, future-version, listener, and ParamValue primitive edges
- StateTree listener, child removal, typed getter, malformed JSON, and synchronizer decode/remove edges
- Binding empty callback and unknown-parameter reset robustness
- EditHistory runtime max-depth trimming and zero-depth behavior
- PropertiesFile scalar parser and malformed JSON shape edges
- PresetManager empty navigation, discovery, PresetInfo load, and rename/current-name edges

Local validation:
- `cmake --build build --target pulp-test-state pulp-test-state-tree pulp-test-binding pulp-test-properties pulp-test-preset-manager pulp-test-edit-history -j8`
- `ctest --test-dir build --output-on-failure -R "ParamRange|ParamValue|StateStore|StateTree|ObservableValue|CachedProperty|PropertiesFile|ApplicationProperties|Binding|PresetManager|EditHistory"` (116/116 passed)
- `git diff --check origin/main...HEAD`

No Namespace/SSH validation dispatched. GitHub-hosted CI is the validation gate.

### CODE REVIEW

File: core/state/src/state_tree_sync.cpp [hunk 1]
```diff
@@ -11,7 +11,9 @@ void StateTreeSynchroniser::attach(StateTree::Ptr tree) {
     listener_id_ = tree_->add_listener(
         [this](StateTree&, std::string_view prop, const PropertyValue&, const PropertyValue& new_val) {
             SyncDelta delta;
-            delta.type = SyncDeltaType::PropertySet;
+            delta.type = std::holds_alternative<std::monostate>(new_val)
+                ? SyncDeltaType::PropertyRemove
+                : SyncDeltaType::PropertySet;
```
[the remark this answers is not in the document]
Comment (danielraffel) [reply core/state/src/state_tree_sync.cpp]: Addressed in e1fc40380: removals are now classified only when the post-notification tree no longer has the property, preserving explicit set(key, PropertyValue{}) as PropertySet. Added a regression covering apply() preserving has("nullable") for explicit null values. Local validation: pulp-test-state-tree build plus StateTree CTest 33/33.

[the remark this answers is not in the document]
Comment (danielraffel) [reply core/state/src/state_tree_sync.cpp]: Addressed on the current branch: the listener now emits PropertyRemove only when the new value is monostate and the node no longer has the property, so explicit PropertyValue{} sets remain PropertySet. The branch also has a regression test, `StateTreeSynchroniser preserves explicit null property sets`, that applies the delta and asserts the key remains present.
